projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/interactions/cartesian-explorer-context-menu/cartesian-explorer-context-menu.component.test.ts

```diff
@@ -0,0 +1,131 @@
+import { ChangeDetectionStrategy, Component, Inject, Injector, Optional } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { TimeRangeService } from '@hypertrace/common';
+import { ButtonComponent, DividerComponent, POPOVER_DATA } from '@hypertrace/components';
+import { createHostFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';
+import { MockComponent } from 'ng-mocks';
+import { CartesianSelectedData, CartesianSeriesVisualizationType } from '../../../../../../../public-api';
+import { CartesainExplorerNavigationService } from '../cartesian-explorer-navigation.service';
+import { CartesianExplorerContextMenuComponent, ContextMenu } from './cartesian-explorer-context-menu.component';
+
+@Component({
+  // tslint:disable-next-line:component-selector
+  selector: 'test-context-menu-content',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<div class="test-modal-content">Test Component Content</div> `
+})
+class TestComponent {
+  public constructor(
+    public readonly injector: Injector,
+    @Optional() @Inject(POPOVER_DATA) public readonly data: unknown
+  ) {}
+}
+
+describe('Sheet Overlay component', () => {
+  const selectedData: CartesianSelectedData<unknown> = {
+    timeRange: TimeRangeService.toFixedTimeRange(
+      new Date('2021-11-02T05:33:19.288Z'),
+      new Date('2021-11-02T14:30:15.141Z')
+    ),
+    selectedData: [
+      {
+        dataPoint: { timestamp: '2021-11-02T05:40:00.000Z', value: 1477.3599999999983 },
+        context: {
+          data: [
+            { timestamp: '2021-11-02T05:15:00.000Z', value: 774 },
+            { timestamp: '2021-11-02T05:40:00.000Z', value: 1477.3599999999983 },
+            { timestamp: '2021-11-02T12:05:00.000Z', value: 1056.48 }
+          ],
+          units: 'ms',
+          color: '#4b5f77',
+          name: 'p99',
+          type: CartesianSeriesVisualizationType.Column,
+          stacking: false,
+          hide: false
+        },
+        location: { x: 59, y: 31.023400000000215 }
+      },
+      {
+        dataPoint: { timestamp: '2021-11-02T12:05:00.000Z', value: 1056.48 },
+        context: {
+          data: [
+            { timestamp: '2021-11-02T05:15:00.000Z', value: 774 },
+            { timestamp: '2021-11-02T05:40:00.000Z', value: 1477.3599999999983 },
+            { timestamp: '2021-11-02T12:05:00.000Z', value: 1056.48 }
+          ],
+          units: 'ms',
+          color: '#4b5f77',
+          name: 'p99',
+          type: CartesianSeriesVisualizationType.Column,
+          stacking: false,
+          hide: false
+        },
+        location: { x: 138, y: 82.58120000000001 }
+      }
+    ],
+    location: { x: 452, y: 763 }
+  };
+  let spectator: Spectator<CartesianExplorerContextMenuComponent<unknown>>;
+
+  const createHost = createHostFactory({
```

Comment:
nit: you can use a host factory, but it's overkill for just testing a component without inputs. Component factory would work with less extra config.

Reply:
Used host factory to inject POPOVER_DATA to the test
![Screenshot 2022-01-17 at 8 25 02 AM](https://user-images.githubusercontent.com/86998924/149701191-8378e832-d234-4e32-ab91-657905cb6a6c.png)